### ISSUE TITLE
Branches are updating incorrectly with the `main` and `next` development flow

### ISSUE THREAD

**Issue by andrii-bodnar** [codecov/codecov-action#912]
We have two development branches in parallel: `main` and `next`.

The problem is that the Codecov GH Action does not correctly update these branches on the Codecov side and the `main` branch doesn't contain all the commits that exist in the `main` branch in the Repo.

We have a separate workflow to sync `main` to `next`. After merging this sync PR, the Codecov is porting all changes to the `next` branch and the `main` branch becomes broken.

For example:

- the `main` branch commits (**broken**):
  - Codecov project - https://app.codecov.io/gh/lingui/js-lingui/commits
  - GitHub repo - https://github.com/lingui/js-lingui/commits/main

- the `next` branch commits (**equal**):
  - Codecov project - https://app.codecov.io/gh/lingui/js-lingui/commits?branch=next
  - GitHub repo - https://github.com/lingui/js-lingui/commits/next

As you can see, the codecov's `main` contains a few latest commits that weren't merged into the `next` yet and after its merging - these commits will disappear from the codecov's `main` and will appear only in the `next` branch. It will contain only those commits that were uploaded more than a month ago by using the v1 version of the action.

This is totally weird and unclear why this happens.

GH Workflow - https://github.com/lingui/js-lingui/blob/main/.github/workflows/main-suite.yml#L76
Sync `main` to `next` Workflow - https://github.com/lingui/js-lingui/blob/main/.github/workflows/next-sync.yml

Hoping for someone's help with this, thanks in advance!

P.S. The reporting WF was broken and did not report for a while in the `main` branch, but since when it was fixed - the codecov's `main` branch is still broken.
